src/Form/AuthenticationForm.php

```diff
@@ -597,4 +500,155 @@ public function ajaxCallback(array $form, FormStateInterface $form_state): array
     return $form;
   }
 
+  /**
+   * Creates a new auth key stored in a file.
+   *
+   * @return \Drupal\key\KeyInterface
+   *   A new auth key.
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   * @throws \Drupal\key\Exception\KeyValueNotSetException
+   */
+  protected function generateNewAuthKey() {
```

Comment:
@cnovak I still disagree with this approach where we do not give the chance for users to save their Apigee Edge credentials an actually secure storage instead of an *unencrypted* file in the private file system. I thought we have made a step forward about this topic in our past conversations. We would just need to display the available key storages on the form, even if we hide them by default and display them in the "advanced" section.

Reply:
We are going to address this in another ticket. We intend to present links to the documentation for advance configuration and give suggestions for best practices.